open-sphere-base/control-panels/src/main/java/io/opensphere/controlpanels/state/StateControllerImpl.java

```diff
@@ -28,20 +43,86 @@
+
+    /** The root data group info. */
+    private final DefaultDataGroupInfo myRootGroupInfo;
+
+    /** The data group controller. */
+    private final DataGroupController myDataController;
+
+    /** The order manager for data type info. */
+    private final DataTypeInfoOrderManager myOrderManager;
+
+    /** The lock to control calls for toggling state. */
+    private boolean myToggleLock;
+
+    /** The toolbox. */
+    private final Toolbox myToolbox;
+
+    /** The activation listener. */
+    private final ActivationListener myActivationListener = new AbstractActivationListener()
+    {
+        @Override
+        public void handleDeactivating(DataGroupInfo dgi)
+        {
+            if (dgi.hasMembers(false))
+            {
+                for (DataTypeInfo dti : dgi.getMembers(false))
+                {
+                    if (isStateActive(dti.getDisplayName()))
+                    {
+                        toggleStateWithLock(dti.getDisplayName());
+                    }
+                }
+            }
+        }
+
+        @Override
+        public boolean handleActivating(DataGroupInfo dgi, io.opensphere.core.util.lang.PhasedTaskCanceller canceller)
+                throws DataGroupActivationException, InterruptedException
+        {
+            boolean stateWasToggled = false;
+            if (dgi.hasMembers(false))
+            {
+                for (DataTypeInfo dti : dgi.getMembers(false))
+                {
+                    if (!isStateActive(dti.getDisplayName()))
+                    {
+                        toggleStateWithLock(dti.getDisplayName());
+                        stateWasToggled = true;
+                    }
+                }
+            }
+            return stateWasToggled;
+        }
+    };
     /**
      * Constructor.
      *
-     * @param moduleStateManager The module state manager.
+     * @param toolbox The toolbox.
      */
-    public StateControllerImpl(ModuleStateManager moduleStateManager)
+    public StateControllerImpl(Toolbox toolbox)
     {
-        myModuleStateManager = Utilities.checkNull(moduleStateManager, "moduleStateManager");
+        myToolbox = toolbox;
+        myModuleStateManager = Utilities.checkNull(myToolbox.getModuleStateManager(), "moduleStateManager");
+        myRootGroupInfo = new DefaultDataGroupInfo(true, myToolbox, "State", "State", "State");
+        myDataController = MantleToolboxUtils.getMantleToolbox(myToolbox).getDataGroupController();
+        myDataController.addRootDataGroupInfo(myRootGroupInfo, this);
+        myOrderManager = new DefaultDataTypeInfoOrderManager(myToolbox.getOrderManagerRegistry());
+        myOrderManager.open();
+        myToggleLock = false;
+        checkInitialStates();
     }
 
     @Override
     public void deactivateAllStates()
     {
-        myModuleStateManager.deactivateAllStates();
+        for (String id : myModuleStateManager.getActiveStateIds())

```

Comment:
do this as a stream (since it's a simple loop just acting on each active state id)